adaptdl/adaptdl/torch/adascale_test.py

```diff
@@ -103,25 +111,22 @@ def rosenbrock(x, y):
 
     sgd = torch.optim.SGD(params_t, lr=0.001)
     schedule = torch.optim.lr_scheduler.MultiStepLR(sgd, [1000])
-    obj = adascale.AdaScale(sgd, scale=1.0, num_replicas=1,
-                            patch_optimizer=True)
-    i = 0.0
-    while (i < 100000 and
-           not (params_t[0]['params'][0].allclose(torch.tensor([1.0]),
-                                                  atol=0.01)
-                and params_t[1]['params'][0].allclose(torch.tensor([1.0]),
-                                                      atol=0.01))):
+    adp = MagicMock()
+    adp.require_backward_grad_sync = True
+    adascale.AdaScale(adp, sgd, accum_scale=1.0, num_replicas=1,
+                      patch_optimizer=True)
+    for i in range(100000):
         sgd.zero_grad()
         loss = rosenbrock(params_t[0]['params'][0], params_t[1]['params'][0])
         loss.backward()
         sgd.step()
-        i += obj.get_progress()
         schedule.step()
-    print(params_t)
-    assert(params_t[0]['params'][0].allclose(torch.tensor([1.0]),
-                                             atol=0.01)
-           and params_t[1]['params'][0].allclose(torch.tensor([1.0]),
-                                                 atol=0.01))
+        if params_t[0]['params'][0].allclose(torch.tensor([1.0]), atol=0.01) \
+                and params_t[1]['params'][0].allclose(torch.tensor([1.0]),
+                                                      atol=0.01):
+            break
+    else:
+        assert False, params_t
```

Comment:
Can you add a print/log that this error is caused by the optimizer not converging in the proper number of steps?